src/main/java/com/faforever/client/fa/relay/ice/IceAdapterImpl.java

```diff
@@ -297,6 +297,7 @@ private void updateLobbyModeFromGameInfo(GameLaunchMessage gameLaunchMessage) {
       return;
     }
 
+    // TODO: Remove this?
```

Comment:
I think it should be if(Matchmaker) right? This is needed though I think. Can u check this?

Reply:
Yea that's what I'm thinking too. We could just get rid of the init mode entirely, and always use autolobby for matchmaker games

Reply:
Yes can u do so @Askaholic and then I merge this...

Reply:
Ok, will do

Reply:
So I looked into this, the problem is we didn't add game_type to the game_launch message, so it's not available in this function. I think it's ok to have the extra check for ladder game mode since it would only apply if for some reason the server didn't send the init mode (shouldn't happen, but it doesn't really hurt to have the code in there).